Resolve Issue #251 - Update Tag component styles to vertically center content

nicklemmon opened the pull request:
Resolves #251 

## What Changed?
- Updated display of tag `.Content` from `inline-block` to `inline-flex`
- Added some horizontal spacing for child SVG elements

**Before:**
![Screen Shot 2019-09-24 at 8 44 57 AM](https://user-images.githubusercontent.com/3613392/65512923-36fe0580-dea8-11e9-84d4-5c9e8f0dced9.png)

**After:**
![Screen Shot 2019-09-24 at 8 59 48 AM](https://user-images.githubusercontent.com/3613392/65513676-b213eb80-dea9-11e9-93df-90fc1e421d28.png)

## How to Test

Look at variants of `<Tag/>` and verify vertical centering:
http://localhost:9001/?selectedKind=Feedback%7CTag&selectedStory=basic%20tag&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

jonambas commented on packages/matchbox/src/components/Tag/Tag.module.scss:
```diff
@@ -114,14 +114,24 @@
+  display: inline-flex;
+  align-items: center;
+  justify-content: space-between;
   vertical-align: middle;
```
Nice this works.

Couple questions - Do we still need `vertical-align: middle`? Looks like align-items takes precedence. And do we need `space-between` when the `.Tag` container is an inline element with no set width?

nicklemmon replied:
Good questions. Let me play with it some more.

nicklemmon commented:
Alright @jonambas - let me know what you think about these tweaks.

I also went ahead and removed that global `transition` property that could cause some perf problems. Same fix we added to the buttons a couple of weeks ago.
